### PR TITLE
Add customSelectedOptionRenderer to style ContextSwitcher dropdown button

### PR DESCRIPTION
**Pull request by ndricimrr**
**Description**

Changes proposed in this pull request:
Added function to customise context switcher button label

In order to test it append the following entry to **contextSwitcher.options**  at  _test/e2e-test-application/src/luigi-config/extended/navigation.js:233_ after **label** and **pathValue**
`customRendererCategory: n !== 1 ? n % 2 == 0  ? "production" : "stage" : “”`

Next add another entry to the  `contextSwitcher{...} ` 
```
...
customSelectedOptionRenderer: (option) => {
    if (option.customRendererCategory === 'production') {
      // render a green bubble
      return `<label style="color: #8F0; font-weight:bold">
      ${option.label}
      <img src="https://i.ibb.co/sJZ4TwX/favicon-16x16.png" height="10" />
      </label>`;
    } else if (option.customRendererCategory === 'stage') {
      // render a blue bubble
      return `<label style="color: #08F; font-weight:bold">
      ${option.label}
      <img src="https://i.ibb.co/4p6jtzw/favicon-16x16.png" height="10"  />
      </label>`;
    } else {
    return `${option.label}`;
    }
},
..
```
Result: 
cswitcher

Add this flag to lerna when bootstrapping if your dependencies are not linked:
`lerna bootstrap --force-local`

**Related issue(s)**

### CODE REVIEW

File: docs/navigation-parameters-reference.md [hunk 1]
```diff
@@ -294,6 +294,11 @@ The context switcher is a drop-down list available in the top navigation bar. It
   - **label** defines the context element label. If not defined, the **pathValue** is passed to **fallbackLabelResolver** to set its value. The default value is **pathValue**, if **fallbackLabelResolver** is not defined.
   - **pathValue** defines the context element path that is appended to **parentNodePath** and reflects a **pathSegment**.
 
+### customSelectedOptionRenderer
+- **type**: function
+- **parameters**: label, isSelected
+- **description**: enables you to customize the dropdown button of the context switcher by rendering custom button label through returning an html string. The function takes a boolean **isSelected** as a parameter, defining the state of the button and a string **label** defining the label of the button.
```
Comment (alexandra-simeonova) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **description**: enables you to customize the dropdown button of the context switcher by rendering a custom button label through the return of a HTML string. The function takes a boolean **isSelected** as a parameter, defining the state of the button, and a string **label** defining the label of the button.
```

File: docs/navigation-advanced.md [hunk 2]
```diff
@@ -263,6 +263,9 @@ contextSwitcher: {
   lazyloadOptions: false,
   fallbackLabelResolver: (id) => (id.toUpperCase()),
   options: [{label,pathValue}, {label,pathValue}],
+  customSelectedOptionRenderer: (label, isSelected)=>{
+      return isSelected ? `<label>${label} test</label>` : `<label>${label} </label>`;
+  },
   customOptionsRenderer: (option, isSelected) => {let className = 'fd-menu__item' + (isSelected ? ' is-selected' : '');
   return `<a class="${className}">${option.label} test</a>`;},
```
Comment (maxmarkus) [inline docs/navigation-advanced.md]: Just slightly better readability.
```suggestion
  customOptionsRenderer: (option, isSelected) => {
    let className = 'fd-menu__item' + (isSelected ? ' is-selected' : '');
    return `<a class="${className}">${option.label} test</a>`;
  },
```

File: docs/navigation-advanced.md [hunk 3]
```diff
@@ -263,6 +263,9 @@ contextSwitcher: {
   lazyloadOptions: false,
   fallbackLabelResolver: (id) => (id.toUpperCase()),
   options: [{label,pathValue}, {label,pathValue}],
+  customSelectedOptionRenderer: (label, isSelected)=>{
+      return isSelected ? `<label>${label} test</label>` : `<label>${label} </label>`;
```
Comment (maxmarkus) [inline docs/navigation-advanced.md]:
```suggestion
  customSelectedOptionRenderer: (label, isSelected) => {
      return isSelected ? `<label class="is-selected">${label} </label>` : `<label>${label} </label>`;
```

File: core/src/navigation/ContextSwitcher.html [hunk 4]
```diff
@@ -153,6 +157,12 @@ <h5 class="fd-action-bar__title fd-has-type-1">
         )
           ? contextSwitcherConfig.customOptionsRenderer
           : undefined;
+
+        customSelectedOptionRenderer = GenericHelpers.isFunction(
```
Comment (maxmarkus) [inline core/src/navigation/ContextSwitcher.html]: Somehow weird intendation, but I fear this is automatically

File: docs/navigation-advanced.md [hunk 5]
```diff
@@ -262,9 +262,20 @@ contextSwitcher: {
...
   fallbackLabelResolver: (id) => (id.toUpperCase()),
-  options: [{label,pathValue}, {label,pathValue}],
-  customOptionsRenderer: (option, isSelected) => {let className = 'fd-menu__item' + (isSelected ? ' is-selected' : '');
-  return `<a class="${className}">${option.label} test</a>`;},
+  options: [{label, pathValue,customRendererCategory}, {label, pathValue, customRendererCategory}],
+  customSelectedOptionRenderer: (option) => {
+    let color = "white";
+    if(option.customRendererCategory === 'production'){
+      color = "#87FF00";
+    }else if (option.customRendererCategory === 'stage'){
```
Comment (maxmarkus) [inline docs/navigation-advanced.md]:
```suggestion
    if (option.customRendererCategory === 'production') {
      color = "#87FF00";
    } else if (option.customRendererCategory === 'stage') {
```

**Pull request merged**